o2g/osm/builders/stations.md

```diff
@@ -0,0 +1,37 @@
+Notes on Stations
+-----------------
+
+Station is a node with "public_transport:station" and "railway:station" tags. could appear in different relations as "stop", "halt" or a simple member node. Stations appear in GTFS stops.txt as parent_stop of other stops. This information will be used to produce "transfers" in trip planning. That's why we need to process them correctly.
+
+Station node is member of a route relation as "stop". In this case it has "public_transport:station" and "railway:station" tags.
+
+Station node is member of a route relation with no relation type. It is just a member. Tags are like the first case.
+
+(hint: other members have "railway:stop" or "railway:platform" tags.)
+
+As an example see Freiburg Hbf. [Station node](https://www.openstreetmap.org/node/21769883) appears in many relations as "stop", "halt" or without membership type. It has "public_transport:staion" and "railway:stop" tags. In [this relation (Freiburg Hbf))](https://www.openstreetmap.org/relation/6311483) it appears as a member node with no type, however in [this one)[https://www.openstreetmap.org/relation/4573043] it appears as a "stop" and in [this one](https://www.openstreetmap.org/relation/70678) as a "halt".
+
+
+Stations, Stop Areas and Stops
+------------------------------
+When we come across stop_area relations, we have to make sure there is one station for that stop_area or we have to create one (in case the station node is missing in OSM or is not linked to any stop_area). For each stop in the stop_are we have to set the correct parent_station, which would be another stop in GTFS jargon.
```

Comment:
hmm, yes. I think OSM defines a lot of different things and GTFS is rather limiting, even for public transport. This aside, GTFS has `stop`, `parent_station` and `entrance/exit` which all go into `stops.txt`. A `stop_area` is not necessarily equivalent to a `parent_station` in GTFS. A `stop_area` could be a train station (like Hamburg Hbf) or a bus terminal (see Hamburg ZOB). In both cases there *might* be a member node which is tagged as `amenity:bus_station` and/or `public_transport:station`. If this node is there we can take its coordinates and put it in `stops.txt` as a `parent_station` for other stops in that `stop_area`. Otherwise we have to use coordinate of other nodes in that stop_area to create the aforementioned record.